File: mobile/android/app/src/main/java/com/navtin/ui/home/HomeView.kt
package com.navtin.ui.home

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedRectangleCornerSize
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Shadow
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.navtin.engine.GameMode
import com.navtin.engine.GameState
import com.navtin.engine.GamePhase
import com.navtin.engine.PlayerTurn
import com.navtin.ui.AppTheme
import com.navtin.ui.board.BoardView

@Composable
fun HomeView() {
    val state = remember { GameState() }
    var showGame by remember { mutableStateOf(false) }
    var showMatchmaking by remember { mutableStateOf(false) }
    
    if (showGame) {
        BoardView(state = state)
    } else if (showMatchmaking) {
        com.navtin.ui.matchmaking.MatchmakingView(
            state = state,
            onMatchFound = { showMatchmaking = false; showGame = true },
            onCancel = { showMatchmaking = false }
        )
    } else {
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(AppTheme.background),
            contentAlignment = Alignment.Center
        ) {
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.SpaceBetween,
                modifier = Modifier.fillMaxHeight().padding(vertical = 60.dp)
            ) {
                // Logo Section
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Text(
                        text = "NAV-TIN",
                        color = AppTheme.neonCyan,
                        fontSize = 64.sp,
                        fontWeight = FontWeight.Bold,
                        style = androidx.compose.ui.text.TextStyle(
                            shadow = Shadow(color = AppTheme.neonCyan, blurRadius = 40f)
                        )
                    )
                    Text(
                        text = "Cyber Strategy",
                        color = Color.White.copy(alpha = 0.6f),
                        fontSize = 16.sp,
                        letterSpacing = 5.sp
                    )
                }
                
                // Buttons
                Column(
                    modifier = Modifier.fillMaxWidth().padding(horizontal = 40.dp),
                    verticalArrangement = Arrangement.spacedBy(20.dp)
                ) {
                    MenuButton("PLAY VS AI", AppTheme.neonCyan) {
                        state.mode = GameMode.VS_AI
                        startGame(state)
                        showGame = true
                    }
                    MenuButton("LOCAL 1V1", AppTheme.neonPink) {
                        state.mode = GameMode.LOCAL
                        startGame(state)
                        showGame = true
                    }
                    MenuButton("ONLINE MATCH", AppTheme.neonBorder) {
                        showMatchmaking = true
                    }
                    MenuButton("SETTINGS", Color.Gray) {}
                }
                
                // Ads Section
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Box(
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(50.dp)
                            .background(Color.White.copy(alpha = 0.05f)),
                        contentAlignment = Alignment.Center
                    ) {
                        Text("AD BANNER", color = Color.White.copy(alpha = 0.3f), fontSize = 10.sp)
                    }

                    Text(
                        text = "VER 1.0.0",
                        color = Color.White.copy(alpha = 0.3f),
                        fontSize = 12.sp,
                        modifier = Modifier.padding(top = 10.dp)
                    )
                }
            }
        }
    }
}

@Composable
fun MenuButton(title: String, color: Color, isDisabled: Boolean = false, onClick: () -> Unit) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .height(60.dp)
            .background(if (isDisabled) Color.Transparent else color.copy(alpha = 0.1f), shape = androidx.compose.foundation.shape.RoundedCornerShape(12.dp))
            .border(2.dp, if (isDisabled) Color.Gray else color, shape = androidx.compose.foundation.shape.RoundedCornerShape(12.dp))
            .clickable(enabled = !isDisabled) { onClick() },
        contentAlignment = Alignment.Center
    ) {
        Text(
            text = title,
            color = if (isDisabled) Color.Gray else Color.White,
            fontWeight = FontWeight.Bold,
            fontSize = 18.sp
        )
    }
}

private fun startGame(state: GameState) {
    state.piecePositions.clear()
    state.player1PiecesOnBoard = 0
    state.player2PiecesOnBoard = 0
    state.player1PiecesLeftToPlace = 9
    state.player2PiecesLeftToPlace = 9
    state.phase = GamePhase.PLACING
    state.turn = PlayerTurn.PLAYER_1
    state.isRemovingPiece = false
    state.message = "Player 1: Place a piece"
}

File: mobile/ios/NavTin/Views/HomeView.swift
import SwiftUI

struct HomeView: View {
    @StateObject var state = GameState()
    @State var showGame = false
    @State var showMatchmaking = false
    
    var body: some View {
        NavigationView {
            ZStack {
                AppTheme.background.ignoresSafeArea()
                
                VStack(spacing: 30) {
                    // Logo Section
                    VStack {
                        Text("NAV-TIN")
                            .font(.custom("Inter-Bold", size: 64))
                            .foregroundColor(AppTheme.neonCyan)
                            .shadow(color: AppTheme.neonCyan, radius: 20)
                        
                        Text("Cyber Strategy")
                            .font(.headline)
                            .foregroundColor(.white.opacity(0.6))
                            .tracking(5)
                    }
                    .padding(.top, 50)
                    
                    Spacer()
                    
                    // Buttons
                    VStack(spacing: 20) {
                        MenuButton(title: "PLAY VS AI", color: AppTheme.neonCyan) {
                            state.mode = .vsAI
                            startGame()
                        }
                        
                        MenuButton(title: "LOCAL 1V1", color: AppTheme.neonPink) {
                            state.mode = .local
                            startGame()
                        }
                        
                        MenuButton(title: "ONLINE MATCH", color: AppTheme.neonBorder) {
                            showMatchmaking = true
                        }
                        
                        MenuButton(title: "SETTINGS", color: .gray) {
                            // Settings
                        }
                    }
                    .padding(.horizontal, 40)
                    
                    Spacer()
                    
                    // Footer
                    Text("VER 1.0.0")
                        .font(.caption2)
                        .foregroundColor(.white.opacity(0.3))
                        .padding(.bottom, 20)
                }
                
                // Banner Ad Placeholder
                VStack {
                    Text("AD BANNER")
                        .font(.caption2)
                        .foregroundColor(.white.opacity(0.3))
                        .frame(maxWidth: .infinity, maxHeight: 50)
                        .background(Color.white.opacity(0.05))
                    
                    Text("VER 1.0.0")
                        .font(.caption2)
                        .foregroundColor(.white.opacity(0.3))
                        .padding(.bottom, 20)
                }
            }
            .onAppear {
                AdMobManager.shared.requestATT()
            }
            .fullScreenCover(isPresented: $showGame) {
                BoardView(state: state)
            }
            .fullScreenCover(isPresented: $showMatchmaking) {
                MatchmakingView(state: state)
            }
        }
    }
    
    private func startGame() {
        // Reset state
        state.piecePositions = [:]
        state.player1PiecesOnBoard = 0
        state.player2PiecesOnBoard = 0
        state.player1PiecesLeftToPlace = 9
        state.player2PiecesLeftToPlace = 9
        state.phase = .placing
        state.turn = .player1
        state.isRemovingPiece = false
        state.message = "Player 1: Place a piece"
        
        showGame = true
    }
}

struct MenuButton: View {
    let title: String
    let color: Color
    var isDisabled: Bool = false
    let action: () -> Void
    
    var body: some View {
        Button(action: action) {
            Text(title)
                .font(.headline.bold())
                .foregroundColor(isDisabled ? .gray : .white)
                .frame(maxWidth: .infinity)
                .padding(.vertical, 20)
                .background(
                    RoundedRectangle(cornerRadius: 12)
                        .stroke(isDisabled ? Color.gray : color, lineWidth: 2)
                        .background(isDisabled ? Color.clear : color.opacity(0.1))
                )
                .shadow(color: isDisabled ? .clear : color, radius: 5)
        }
        .disabled(isDisabled)
    }
}
